sbhs/views.py

```diff
@@ -318,15 +318,31 @@ def slot_new(request):
             form = SlotCreationForm(request.POST)
             if form.is_valid():
                 new_slot = form.save(commit=False)
-
                 new_slot_date = new_slot.start_time.date()
                 new_slot_time = new_slot.start_time.time()
+                prev_hour = dt.combine(
+                            new_slot_date,time(
+                                new_slot_time.hour,00)) - timedelta(hours=1)
+                prev_slot = slot_history.filter(start_time__date=new_slot_date,
+                                                start_time__time=prev_hour)
+                next_hour = dt.combine(
+                                new_slot_date,time(
+                                new_slot_time.hour,00)) + timedelta(hours=1)
+                next_slot = slot_history.filter(start_time__date=new_slot_date,
+                                                start_time__time=next_hour)
+                if prev_slot.exists() or next_slot.exists():
```

Comment:
Slight not clear how this if works!
Can you please explain.
B'coz
1. Will next slot ever exists?
2. prev_slot is exact match with the hour!

Also, better would be writing tests, but as this needs to go live ASAP so kindly check this with all the corner cases manually once.

Reply:
Next slot exists if you have booked future slot in advance.